plugins/analysis/dashboard/fpc_livefeed.go

```diff
@@ -91,61 +93,76 @@ func createFPCUpdate(hb *packet.FPCHeartbeat, recordEvent bool) *FPCUpdate {
-			// update recorded events
-			recordedConflicts.update(ID, conflict{NodesView: map[string]voteContext{nodeID: newVoteContext}})
-		}
+		// update recorded events
+		activeConflicts.update(ID, conflict{NodesView: map[string]voteContext{nodeID: newVoteContext}})
 	}
 
-	if recordEvent {
-		// check finalized conflicts
-		if len(hb.Finalized) > 0 {
-			finalizedConflicts := make([]FPCRecord, len(hb.Finalized))
-			i := 0
-			for ID, finalOpinion := range hb.Finalized {
-				conflictOverview, ok := recordedConflicts.load(ID)
-				if !ok {
-					log.Error("Error: missing conflict with ID:", ID)
-					continue
-				}
-				conflictDetail := conflictOverview.NodesView[nodeID]
-				conflictDetail.Status = vote.ConvertOpinionToInt32(finalOpinion)
-				conflicts[ID] = newConflict()
-				conflicts[ID].NodesView[nodeID] = conflictDetail
-				recordedConflicts.update(ID, conflicts[ID])
-				finalizedConflicts[i] = FPCRecord{
-					ConflictID: ID,
-					NodeID:     conflictDetail.NodeID,
-					Rounds:     conflictDetail.Rounds,
-					Opinions:   conflictDetail.Opinions,
-					Status:     conflictDetail.Status,
-				}
-				i++
-
-				metrics.Events().AnalysisFPCFinalized.Trigger(&metrics.AnalysisFPCFinalizedEvent{
-					ConflictID: ID,
-					NodeID:     conflictDetail.NodeID,
-					Rounds:     conflictDetail.Rounds,
-					Opinions:   vote.ConvertInts32ToOpinions(conflictDetail.Opinions),
-					Status:     vote.ConvertInt32Opinion(conflictDetail.Status),
-				})
-			}
+	// check finalized conflicts
+	if len(hb.Finalized) == 0 {
+		return &FPCUpdate{Conflicts: conflicts}
+	}
 
-			err := storeFPCRecords(finalizedConflicts, mongoDB())
-			if err != nil {
-				log.Errorf("Error while writing on MongoDB: %s", err)
-			}
+	finalizedConflicts := make(FPCRecords, len(hb.Finalized))
+	i := 0
+	for ID, finalOpinion := range hb.Finalized {
+		conflictOverview, ok := activeConflicts.load(ID)
+		if !ok {
+			log.Error("Error: missing conflict with ID:", ID)
+			continue
+		}
+		conflictDetail := conflictOverview.NodesView[nodeID]
+		conflictDetail.Outcome = vote.ConvertOpinionToInt32(finalOpinion)
+		conflicts[ID] = newConflict()
+		conflicts[ID].NodesView[nodeID] = conflictDetail
+		activeConflicts.update(ID, conflicts[ID])
+		finalizedConflicts[i] = FPCRecord{
+			ConflictID: ID,
+			NodeID:     conflictDetail.NodeID,
+			Rounds:     conflictDetail.Rounds,
+			Opinions:   conflictDetail.Opinions,
+			Outcome:    conflictDetail.Outcome,
 		}
+		i++
+
+		metrics.Events().AnalysisFPCFinalized.Trigger(&metrics.AnalysisFPCFinalizedEvent{
+			ConflictID: ID,
+			NodeID:     conflictDetail.NodeID,
+			Rounds:     conflictDetail.Rounds,
+			Opinions:   vote.ConvertInts32ToOpinions(conflictDetail.Opinions),
+			Outcome:    vote.ConvertInt32Opinion(conflictDetail.Outcome),
+		})
 	}
 
-	return &FPCUpdate{
-		Conflicts: conflicts,
+	if !config.Node.GetBool(CfgMongoDBEnabled) {
+		return &FPCUpdate{Conflicts: conflicts}
 	}
+
+	// store FPC records on DB.
+	go func() {
```

Comment:
Don't run this in a separate goroutine.

Reply:
I've tried but the problem of not running that on a separate goroutine is that if, for any reason the following operation gets slow (we have cancel timeout of 5 seconds there), the entire dashboard gets stuck since the workerpool is basically managing all of the jobs.